src/TinyRAM/EntryPoint.hs

```diff
@@ -118,10 +122,18 @@ appendProgramFile :: ProgramFilePath -> ByteString -> IO ()
 appendProgramFile (ProgramFilePath path) value =
   BS.appendFile path (value <> "\n")
 
+writeProgramFile :: ProgramFilePath -> ByteString -> IO ()
+writeProgramFile (ProgramFilePath path) value =
+  BS.writeFile path (value <> "\n")
```

Comment:
Nit: This is not an atomic nor durable write, so a full disk may leave the FS in an invalid state with a zero-byte file.